velox/connectors/hive/HiveConnector.h

```diff
@@ -92,46 +96,130 @@ class HiveTableHandle : public ConnectorTableHandle {
   const core::TypedExprPtr remainingFilter_;
 };
 
+/// Location related properties of the Hive table to be written
+class LocationHandle {
+ public:
+  enum class TableType {
+    kNew, // Write to a new table to be created.
+    kExisting, // Write to an existing table.
+    kTemporary, // Write to a temporary table.
+  };
+
+  enum class WriteMode {
+    // Write to a staging directory and then move to the target directory
+    // after write finishes.
+    kStageAndMoveToTargetDirectory,
+    // Directly write to the target directory to be created.
+    kDirectToTargetNewDirectory,
```

Comment:
So the difference of kDirectToTargetNewDirectory and kDirectToTargetExistingDirectory is that the table writer needs to create a new directory for the former but expect an existing one for the latter? So if the target directory exists in case of kDirectToTargetNewDirectory, the table write will fail? Could you clarify such semantics in comments? Thanks!

Also can we infer write mode about new or existing targets from table type? Like if we don't do staging, then we always go with kDirectToTargetNewDirectory for new table type but kDirectToTargetExistingDirectory for existing table? If that is the case, then how about we just define two write modes? kStaging and kDirect?